webconsole/v1/types.go

```diff
@@ -0,0 +1,115 @@
+	// If not specified, the built-in logout page is shown.
+	LogoutURL string `json:"logoutURL" protobuf:"bytes,3,opt,name=logoutURL"`
+
+	// MasterPublicURL is how the web console can access the OpenShift v1 server
+	MasterPublicURL string `json:"masterPublicURL" protobuf:"bytes,4,opt,name=masterPublicURL"`
+
+	// LoggingPublicURL is the public endpoint for logging (optional)
+	LoggingPublicURL string `json:"loggingPublicURL" protobuf:"bytes,5,opt,name=loggingPublicURL"`
+
+	// MetricsPublicURL is the public endpoint for metrics (optional)
+	MetricsPublicURL string `json:"metricsPublicURL" protobuf:"bytes,6,opt,name=metricsPublicURL"`
+
+	// ExtensionScripts are file paths on the asset server files to load as scripts when the Web
+	// Console loads
+	ExtensionScripts []string `json:"extensionScripts" protobuf:"bytes,7,rep,name=extensionScripts"`
+
+	// ExtensionProperties are key(string) and value(string) pairs that will be injected into the console under
+	// the global variable OPENSHIFT_EXTENSION_PROPERTIES
+	ExtensionProperties map[string]string `json:"extensionProperties" protobuf:"bytes,8,rep,name=extensionProperties"`
+
+	// ExtensionStylesheets are file paths on the asset server files to load as stylesheets when
+	// the Web Console loads
+	ExtensionStylesheets []string `json:"extensionStylesheets" protobuf:"bytes,9,rep,name=extensionStylesheets"`
+
+	// Extensions are files to serve from the asset server filesystem under a subcontext
+	Extensions []AssetExtensionsConfig `json:"extensions" protobuf:"bytes,10,rep,name=extensions"`
+
+	// ExtensionDevelopment when true tells the asset server to reload extension scripts and
+	// stylesheets for every request rather than only at startup. It lets you develop extensions
+	// without having to restart the server for every change.
+	ExtensionDevelopment bool `json:"extensionDevelopment" protobuf:"varint,11,opt,name=extensionDevelopment"`
+}
+
+// AssetExtensionsConfig holds the necessary configuration options for asset extensions
+type AssetExtensionsConfig struct {
+	// SubContext is the path under /<context>/extensions/ to serve files from SourceDirectory
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// SourceDirectory is a directory on the asset server to serve files under Name in the Web
+	// Console. It may have nested folders.
+	SourceDirectory string `json:"sourceDirectory" protobuf:"bytes,2,opt,name=sourceDirectory"`
+	// HTML5Mode determines whether to redirect to the root index.html when a file is not found.
+	// This is needed for apps that use the HTML5 history API like AngularJS apps with HTML5
+	// mode enabled. If HTML5Mode is true, also rewrite the base element in index.html with the
+	// Web Console's context root. Defaults to false.
+	HTML5Mode bool `json:"html5Mode" protobuf:"varint,3,opt,name=html5Mode"`
+}
+
+// HTTPServingInfo holds configuration for serving HTTP
+type HTTPServingInfo struct {
```

Comment:
This is definitely a common type, but adding it here does not precluding moving it later.  I suspect we'll have a general `config.openshift.io` to hold information like this, but it isn't strictly needed here.